### DISCUSSION TITLE
Where can I learn more on how to alter GNOME UI and make widgets?

### DISCUSSION THREAD

**Discussion by ghxstweb** [material-shell/material-shell#995] [General]
Hi @PapyElGringo ! Thanks for your work on this application. I wanted to ask if there is a site/tutorial or place you recommend newcomers look if they want to get involved in altering GNOME's shell UI, particularly in the realm of designing widgets. I would be very interested in learning more on how to make integrated shell widgets if possible, sort of like what KDE offers. Thanks in advance for your time!

**Comment by PapyElGringo** [comment]
Hey @ghxstbox the only good ressources are available at https://gjs.guide/extensions/ you should also join an ask there https://matrix.to/#/#extensions:gnome.org !

I can personally recommend you to not try making huge alteration on Gnome  but instead work on making additions it's won't break that often !
